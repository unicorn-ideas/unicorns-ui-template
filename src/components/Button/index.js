import React from 'react';
import '../../index.css';

export default ({ onClick, children }) => (
  <button
    className="rounded py-1 px-2 bg-primary-500 text-primary-900 shadow-md hover:bg-secondary-500 hover:text-secondary-100 focus:outline-none focus:shadow-outline"
    onClick={onClick}
  >
    {children}
  </button>
);
