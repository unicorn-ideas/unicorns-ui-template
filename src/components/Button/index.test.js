import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import Button from './index';

test('renders a control', () => {
  const { container, getByText } = render(
    <Button>The Fuolornis Fire Dragons</Button>,
  );
  expect(getByText('The Fuolornis Fire Dragons')).toBeInTheDocument();
});

test('snapshot', () => {
  const tree = renderer
    .create(<Button>The Fuolornis Fire Dragons</Button>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
