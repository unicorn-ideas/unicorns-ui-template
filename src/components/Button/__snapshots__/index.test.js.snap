// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<button
  className="rounded py-1 px-2 bg-primary-500 text-primary-900 shadow-md hover:bg-secondary-500 hover:text-secondary-100 focus:outline-none focus:shadow-outline"
>
  The Fuolornis Fire Dragons
</button>
`;
